sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ColumnResolutionHelper.scala

```diff
@@ -167,19 +167,17 @@ trait ColumnResolutionHelper extends Logging with DataTypeErrorsBase {
           }
         }
 
-        case u @ UnresolvedExtractValue(child, field) =>
+        case u @ UnresolvedExtractValue(child, fieldName) =>
           val newChild = innerResolve(child, isTopLevel = false)
-          val resolvedField = if (conf.getConf(SQLConf.PREFER_COLUMN_OVER_LCA_IN_ARRAY_INDEX)) {
-            innerResolve(field, isTopLevel = false)
-          } else {
-            field
-          }
           if (newChild.resolved) {
-            ExtractValue(child = newChild, extraction = resolvedField, resolver = resolver)
+            ExtractValue(newChild, fieldName, resolver)
           } else {
-            u.copy(child = newChild, extraction = resolvedField)
+            u.copy(child = newChild)
           }
 
+        case d @ DefaultValueExpression(c: Expression, _, _) =>
```

Comment:
I assume this works because we resolve expressions top to bottom, hence we see DefaultValueExpression before we see the unresolved attribute?